superset/views/core.py

```diff
@@ -2464,30 +2463,9 @@ def results(self, key):
     @has_access_api
     @expose("/stop_query/", methods=["POST"])
     @event_logger.log_this
-    @backoff.on_exception(
-        backoff.constant,
-        Exception,
-        interval=1,
-        on_backoff=lambda details: db.session.rollback(),
-        on_giveup=lambda details: db.session.rollback(),
-        max_tries=5,
-    )
     def stop_query(self):
         client_id = request.form.get("client_id")
-
-        query = db.session.query(Query).filter_by(client_id=client_id).one()
-        if query.status in [
-            QueryStatus.FAILED,
-            QueryStatus.SUCCESS,
-            QueryStatus.TIMED_OUT,
-        ]:
-            logging.error(
-                f"Query with client_id {client_id} could not be stopped: query already complete"
-            )
-            return self.json_response("OK")
-        query.status = QueryStatus.STOPPED
-        db.session.commit()
-
+        sql_lab.stop_query.delay(client_id)
```

Comment:
There may not be a message queue, setting up Celery should be optional to simplify installation in modest or early environments